manifests/config/context/resources.pp

```diff
@@ -0,0 +1,77 @@
+# @summary Configure Resources elements in $CATALINA_BASE/conf/context.xml
+#
+# @param ensure
+#   Specifies whether you are trying to add or remove the Resources element.
+# @param resources_name
+#   The name of the Resources to be created, relative to the `java:comp/env` context. `$name`.
+# @param name
+#   `$resources_name`
+# @param resources_type
+#   The fully qualified Java class name expected by the web application when it performs a lookup for this resources. Required to create the resource.
+# @param catalina_base
+#   Specifies the root of the Tomcat installation.
+# @param additional_attributes
+#   Specifies any additional attributes to add to the Valve. Should be a hash of the format 'attribute' => 'value'. Optional
+# @param attributes_to_remove
+#   Specifies an array of attributes to remove from the element. Valid options: an array of strings. `[]`.
+# @param show_diff
+#   Specifies display differences when augeas changes files, defaulting to true. Valid options: true or false.
+#
+define tomcat::config::context::resources (
+  Enum['present','absent'] $ensure = 'present',
+  $resources_name                  = $name,
+  $resources_type                  = undef,
+  $catalina_base                   = $::tomcat::catalina_home,
+  Hash $additional_attributes      = {},
+  Array $attributes_to_remove      = [],
+  Boolean $show_diff               = true,
+) {
+  if versioncmp($::augeasversion, '1.0.0') < 0 {
+    fail('Server configurations require Augeas >= 1.0.0')
+  }
+
+  if $resources_name {
```

Comment:
Unsure why this bit of code is necessary?
Since `$resources_name` defaults to `$name` if not given won't this always be true?